sound/soc/sof/ipc4-topology.c

```diff
@@ -671,8 +671,13 @@ static int sof_ipc4_widget_setup_pcm(struct snd_sof_widget *swidget)
 				      swidget->tuples,
 				      swidget->num_tuples, sizeof(u32), 1);
 		/* Set default DMA buffer size if it is not specified in topology */
-		if (!sps->dsp_max_burst_size_in_ms)
-			sps->dsp_max_burst_size_in_ms = SOF_IPC4_MIN_DMA_BUFFER_SIZE;
+		if (!sps->dsp_max_burst_size_in_ms) {
+			struct snd_sof_widget *pipe_widget = swidget->spipe->pipe_widget;
+			struct sof_ipc4_pipeline *pipeline = pipe_widget->private;
+
+			sps->dsp_max_burst_size_in_ms = pipeline->use_chain_dma ?
+				SOF_IPC4_CHAIN_DMA_BUFFER_SIZE : SOF_IPC4_MIN_DMA_BUFFER_SIZE;
```

Comment:
Not related to this PR, but for all current IPC4 platforms, max burst size is 4ms:
https://github.com/thesofproject/sof/blob/main/zephyr/lib/dma.c#L21

We can't currently query this value (would need to be on a per PCM basis). But if we start hardcoding the actual size for chain-dma in driver (now kernel+FW need to stay aligned), why not do the same for other types of PCMs as well?

Reply:
Hrm, the host buffer size is configured by IPC and it is 2ms by default:

https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/ipc4-topology.h#L74
https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/ipc4-topology.c#L2394

and in FW:
https://github.com/thesofproject/sof/blob/main/src/audio/host-zephyr.c#L934

But are you saying that the 2ms which we set is ignored by fw it is using 4ms instead out of fun? Since when this is done?